saleor/graphql/attribute/filters.py

```diff
@@ -38,7 +39,12 @@ def filter_attributes_by_product_types(qs, field, value, requestor, channel_slug
     if not value:
         return qs
 
-    product_qs = models.Product.objects.visible_to_user(requestor, channel_slug)
+    channel = None
+    if channel_slug is not None:
+        channel = Channel.objects.filter(slug=str(channel_slug)).first()
+    product_qs = models.Product.objects.visible_to_user(
+        requestor, channel, False if channel_slug is None else True
```

Comment:
Is it expected that `if channel_slug is None` will pass if `channel_slug` will be `""`?

Reply:
Yes, it's expected. I don't want to introduce any change in the behavior of this API. In the current 3.14 API return all products if `channel_slug` is not passed for the staff user with product permission. If the same user puts `""` as the channel slug the query will try to return only products available for the channel with `""` as a slug.